mantidproject/mantid
# SANS - Wav loops speedup

**Description of work.**

The final part of the wavelength loops work. In this PR we:
- Move wavelength information into a simpler format (list of pairs)
- Rework the core SANS Reduction loop to process n wavelength slices during one pass, rather than n passes
- "Re-plumb" the properties to get the expected workspaces back, in the expected order
- Simplified the output bundles data structure and how to iterate over it significantly

One victim of these changes was the SANSReductionCore V2 workflow which will no longer work. However this is not used in the group as there is a small unexplained difference in results.
The development effort for the small performance gain on event workspaces to both explain or fix the differences / switch over and now fix-up is not a priority based on the sprint planning.

**To test:**
- Download the ISIS Training data set
- Open Mantid and SANS->ISIS SANS
- Point the user file to either the toml or .txt file called MaskFile in LoqDemo
- Open the CSV file in the same folder
- Select the first row and hit process
- Go to settings, Q, Wavelength
- Under Wavelength change `Log` to `RangeLog`
- Enter `1:1:15` to process wavelength slice 1-15 in 1 Angstrom slices
- It should be noticeably faster than master, data validity has been checked by running the same system tests (unmodified) before and after these changes



Fixes #28987 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
